mail.py

```diff
@@ -0,0 +1,103 @@
+from himlarcli.mail import Mail
+from email.mime.text import MIMEText
+
+
+himutils.is_virtual_env()
+
+parser = Parser()
+options = parser.parse_args()
+printer = Printer(options.format)
+
+ksclient = Keystone(options.config, debug=options.debug)
+ksclient.set_dry_run(options.dry_run)
+logger = ksclient.get_logger()
+novaclient = Nova(options.config, debug=options.debug, log=logger)
+
+# Update these before running the script
+emails_file = 'notify/user_emails.txt'
+content = 'notify/mailto_all.txt'
+subject = 'INFO UH-IaaS'
+
+if hasattr(options, 'region'):
+    regions = ksclient.find_regions(region_name=options.region)
+else:
+    regions = ksclient.find_regions()
+
+if not regions:
+    himutils.sys_error('no valid regions found!')
+
+if 'host' in options and options.host:
+    if '.' in options.host:
+        host = options.host
+    else:
+        domain = ksclient.get_config('openstack', 'domain')
+        host = options.host + '.' + domain
+else:
+    host = None
+
+def action_file():
+    if options.template:
+        content = options.template
+        email_content = open(content, 'r')
+        body_content = email_content.read()
+        if options.dry_run:
+            print content
+            #logger.debug('=> DRY-RUN: print out the content %s' % content)
+        else:
+            mail = Mail(options.config, debug=options.debug)
+            mail.set_dry_run(options.dry_run)
+            msg = MIMEText(body_content, 'plain')
+            msg['subject'] = subject
+            print msg
+            with open(emails_file, 'r') as emails:
+                for toaddr in emails.readlines():
+                    try:
+                        logger.debug('=> Sending email ...')
+                        mail.send_mail(toaddr, msg, fromaddr='noreply@uh-iaas.no')
+                    except ValueError:
+                        himutils.sys_error('Not able to send the email.')
+            emails.close()
+    email_content.close()
+
+
+# Check nova version
+def action_instance():
+    email_content = open(content, 'r')
+    body_content = email_content.read()
+    for region in regions:
+        novaclient = Nova(options.config, debug=options.debug, log=logger) #, region=region)
+        instances = novaclient.get_instances()
+        mail = Mail(options.config, debug=options.debug)
+        mail.set_keystone_client(ksclient)
+        users = mail.mail_instance_owner(instances=instances,
+                                           body=body_content,
+                                           subject=subject,
+                                           admin=True)
+    email_content.close()
+
+
+def action_project():
+    email_content = open(content, 'r')
+    body_content = email_content.read()
+    mail = Mail(options.config, debug=options.debug)
+    #if options.type:
+    #    search_filter['type'] = options.type
+    projects = ksclient.get_projects(domain=options.domain) #, **search_filter)
```

Comment:
Dette burde fungere for å filtrere på project type. dersom du setter --type demo så skal du kun sende mail til dem med instanser i demo prosjekt.